src/rockstor/system/osi.py

```diff
@@ -1203,6 +1203,30 @@ def set_system_rtc_wake(wakeup_epoch):
     return None
 
 
+def is_network_device_responding(address):
+    """
+    Small function that sends ICMP echo requests to a given network device,
+    either using the hostname or IP address.
+    :param address: the hostname or IP address of the device to be pinged
+    :return: true if the device is responding, false if not
+    """
+    # because of -c 3, three requests will be sent
+    output, error, return_code = run_command([PING, '-c 3 -q', address], log=False, throw=False)
+    # The ping command will always return 0 if at least one request has been answered,
+    # 1, if no request has been answered and another error code if the hostname is not known.
+    # The corresponding error message is 'ping: <device name>: Name or service not known'.
+    if return_code == 0:
+        return True
+    elif return_code == 1 or \
+            next((s for s in output if '0 received' in s), None) or \
+            next((s for s in error if 'ping: ' + str(address) in s), None):
+        return False
+    logger.debug('Ping command unexpectedly exited with return code ' + str(return_code))
+    if len(error):
```

Comment:
Small style reformatting
```suggestion
    if len(e):
```